experimenter/experimenter/experiments/models.py

```diff
@@ -835,8 +836,61 @@ def clone(self, name, user, rollout_branch_slug=None):
 
         return cloned
 
-    def get_changelogs(self):
-        return self.changes.all()
+    def get_changelogs_by_date(self):
+        changes_by_date = defaultdict(list)
+        date_option = "%I:%M:%S %p"
+        changelogs = list(
+            self.changes.order_by("-changed_on").prefetch_related("changed_by")
+        )
+
+        for index, changelog in enumerate(changelogs[:-1]):
+            current_data = changelog.experiment_data
+            previous_data = changelogs[index + 1].experiment_data
+            timestamp = changelog.changed_on.strftime(date_option)
+
+            diff_fields = {
+                field: {
+                    "old_value": previous_data.get(field),
+                    "new_value": current_data.get(field),
+                }
+                for field in current_data
+                if (
+                    field != "_updated_date_time"
+                    and field != "published_dto"
+                    and current_data[field] != previous_data.get(field)
+                )
+            }
+
+            for field, field_diff in diff_fields.items():
+                change = {
+                    "field": field,
+                    "event": "GENERAL",
+                    "old_value": field_diff["old_value"],
+                    "new_value": field_diff["new_value"],
+                    "changed_by": changelog.changed_by.get_full_name(),
+                    "timestamp": timestamp,
+                }
+
+                changes_by_date[changelog.changed_on.date()].append(change)
+
+        if creation_log := changelogs[-1] if changelogs else None:
```

Comment:
Having a Python ternary inside an if is weird, I've never seen that before.

I would probably format this as:

```python
if changelogs:
  creation_log = changelogs[-1]
  # ...
```

I'm assuming that `changelogs[-1]` won't be falsy -- it shouldn't need a second if to check it.